File: php_examples/authentication.php
<?php

$secretKey = 'enter your secret key';
$publicKey = 'enter your public key';
$timestamp = time();
$payload = $timestamp . '.' . $publicKey;
$hash = hash_hmac('sha256', $payload, $secretKey, true);
$keys = unpack('H*', $hash);
$hexHash = array_shift($keys);
$signature = $payload . '.' . $hexHash;

$tickerUrl = "https://apiv2.bitcoinaverage.com/indices/global/ticker/BTCUSD"; // request URL
$aHTTP = array(
  'http' =>
    array(
    'method'  => 'GET',
  	)
);
$aHTTP['http']['header']  = "X-Signature: " . $signature;
$context = stream_context_create($aHTTP);
$content = file_get_contents($tickerUrl, false, $context);

echo $content;

?>

File: ruby_examples/authentication.rb
require 'openssl'
require 'date'
require 'open-uri'

public_key = 'enter your public key'

ticker_url = 'https://apiv2.bitcoinaverage.com/indices/global/ticker/BTCUSD'
response = open(ticker_url, 'x-ba-key' => public_key).read
puts response